purchase_last_price_info/models/product.py

```diff
@@ -14,16 +11,22 @@ def _get_last_purchase(self):
         """ Get last purchase price, last purchase date and last supplier """
         lines = self.env['purchase.order.line'].search(
             [('product_id', '=', self.id),
-             ('state', 'in', ['confirmed', 'done'])]).sorted(
+             ('state', 'in', ['purchase', 'done'])]).sorted(
             key=lambda l: l.order_id.date_order, reverse=True)
         self.last_purchase_date = lines[:1].order_id.date_order
         self.last_purchase_price = lines[:1].price_unit
         self.last_supplier_id = lines[:1].order_id.partner_id
```

Comment:
The `search` and `sorted` could be replaced by:
`self.env['purchase.order.line'].search(
            [('product_id', '=', self.id),
             ('state', 'in', ['purchase', 'done'])],
            order='date_order DESC', limit=1)`

Then you can change `lines[:1]` to `lines`.

Also you can take the `date_order` and `partner_id` from lines directly.